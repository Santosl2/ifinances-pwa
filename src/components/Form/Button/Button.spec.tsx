/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable unused-imports/no-unused-imports */
import * as React from "react";

import { render, screen } from "@testing-library/react";

import { Button } from ".";

describe("Button Test", () => {
  it("should render Button", () => {
    render(<Button>Teste</Button>);

    expect(screen.getByText("Teste")).toBeInTheDocument();
  });

  it("should render loading when button is login", () => {
    render(<Button isLoading>Teste</Button>);

    expect(screen.queryByText("Teste")).not.toBeInTheDocument();
  });

  it("should render button with correct color", () => {
    render(<Button bgColor="red">Teste</Button>);

    expect(screen.getByText("Teste")).toHaveStyle({
      background: "red",
    });
  });
});
