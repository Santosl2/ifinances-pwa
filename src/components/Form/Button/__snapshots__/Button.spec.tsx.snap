// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test should match Snapshot 1`] = `
.c0 {
  cursor: pointer;
  background: red;
  color: white;
  border: none;
  border-radius: 2px;
  padding: 0.5rem;
  -webkit-transition: background 0.2s;
  transition: background 0.2s;
}

.c0:hover {
  background: #e60000;
}

.c0:disabled {
  opacity: 0.6;
  cursor: default;
}

<button
  class="c0"
>
  Teste
</button>
`;
